File: dotnet/packages/Microsoft.TeamsAI/Microsoft.TeamsAI/AI/Embeddings/AzureOpenAIEmbeddingsOptions.cs
using Microsoft.Teams.AI.Utilities;

namespace Microsoft.Teams.AI.AI.Embeddings
{
    /// <summary>
    /// Options for configuring an `OpenAIEmbeddings` to generate embeddings using an Azure OpenAI hosted model.
    /// </summary>
    public class AzureOpenAIEmbeddingsOptions : BaseOpenAIEmbeddingsOptions
    {
        /// <summary>
        /// API key to use when making requests to Azure OpenAI.
        /// </summary>
        public string AzureApiKey { get; set; }

        /// <summary>
        /// Name of the Azure OpenAI deployment (model) to use.
        /// </summary>
        public string AzureDeployment { get; set; }

        /// <summary>
        /// Deployment endpoint to use.
        /// </summary>
        public string AzureEndpoint { get; set; }

        /// <summary>
        /// Optional. Version of the API being called.
        /// </summary>
        public string? AzureApiVersion { get; set; }

        /// <summary>
        /// Initializes a new instance of the <see cref="AzureOpenAIEmbeddingsOptions"/> class.
        /// </summary>
        /// <param name="azureApiKey">API key to use when making requests to Azure OpenAI.</param>
        /// <param name="azureDeployment">Name of the Azure OpenAI deployment (model) to use.</param>
        /// <param name="azureEndpoint">Deployment endpoint to use.</param>
        public AzureOpenAIEmbeddingsOptions(
            string azureApiKey,
            string azureDeployment,
            string azureEndpoint) : base()
        {
            Verify.ParamNotNull(azureApiKey);
            Verify.ParamNotNull(azureDeployment);
            Verify.ParamNotNull(azureEndpoint);

            azureEndpoint = azureEndpoint.Trim();

            this.AzureApiKey = azureApiKey;
            this.AzureDeployment = azureDeployment;
            this.AzureEndpoint = azureEndpoint;
        }
    }
}

File: python/packages/ai/teams/ai/embeddings/azure_openai_embeddings_options.py
"""
Copyright (c) Microsoft Corporation. All rights reserved.
Licensed under the MIT License.
"""

from __future__ import annotations

from dataclasses import dataclass
from typing import Dict, List, Optional


@dataclass
class AzureOpenAIEmbeddingsOptions:
    """
    Options for configuring an `AzureOpenAIEmbeddings` to generate embeddings.
    """

    azure_api_key: str
    "API key to use when making requests to Azure OpenAI."

    azure_endpoint: str
    "Deployment endpoint to use."

    azure_deployment: str
    "Name of the Azure OpenAI deployment (model) to use."

    azure_api_version: Optional[str] = None
    "Optional. Version of the API being called. Defaults to `2023-05-15`."

    log_requests: Optional[bool] = False
    "Whether to log requests to the console, useful for debugging and defaults to `false`"

    retry_policy: Optional[List[int]] = None
    "Optional. Retry policy to use in seconds. The default retry policy is `[2, 5]`."

    request_config: Optional[Dict[str, str]] = None
    "Request options to use."
